cli/src/cli_util.rs

```diff
@@ -1613,13 +1613,30 @@ See https://martinvonz.github.io/jj/latest/working-copy/#stale-working-copy \
         }
 
         let settings = self.settings();
-        if settings.user_name().is_empty() || settings.user_email().is_empty() {
+        let missing_user_name = settings.user_name().is_empty();
+        let missing_user_mail = settings.user_email().is_empty();
+        if missing_user_name || missing_user_mail {
+            let mut writer = ui.warning_default();
             writeln!(
-                ui.warning_default(),
-                r#"Name and email not configured. Until configured, your commits will be created with the empty identity, and can't be pushed to remotes. To configure, run:
-  jj config set --user user.name "Some One"
-  jj config set --user user.email "someone@example.com""#
+                writer,
+                "{} not configured. Until configured, your commits will be created with the empty \
+                 identity, and can't be pushed to remotes. To configure, run:",
+                match (missing_user_name, missing_user_mail) {
+                    (true, true) => "Name and email",
+                    (true, false) => "Name",
+                    (false, true) => "Email",
+                    _ => unreachable!(),
```

Comment:
nit: can you split by sentence (to help grepping and possible i18n support)?
```rust
match .. {
    (true, true) => "Name and email not configured.",
    ...
}
```
btw, I prefer not inline large code block as expression.

Reply:
My suggestion would've been to extract it into a variable above the usage.